### PR TITLE
Bump `pytest-mypy-plugins` and improve `jinja2` stubs

### PR DESCRIPTION
**Pull request by sobolevn**
Refs https://github.com/typeddjango/pytest-mypy-plugins/issues/118

### CODE REVIEW

File: mypy_django_plugin/django/context.py [hunk 1]
```diff
@@ -60,12 +60,11 @@ def initialize_django(settings_module: str) -> Tuple["Apps", "LazySettings"]:
         from django.apps import apps
         from django.conf import settings
 
-        apps.get_models.cache_clear()  # type: ignore
         apps.get_swappable_settings_name.cache_clear()  # type: ignore
+        apps.clear_cache()
```
Comment (sobolevn) [inline mypy_django_plugin/django/context.py:64]: Btw, why `apps.get_swappable_settings_name.cache_clear()` is not a part of `app.clear_cache()`? Should I commit this to Django?

Comment (adamchainz) [reply mypy_django_plugin/django/context.py:64]: Make a ticket!

Comment (sobolevn) [reply mypy_django_plugin/django/context.py:64]: Done https://code.djangoproject.com/ticket/34687#ticket

File: django-stubs/template/backends/jinja2.pyi [hunk 2]
```diff
@@ -17,4 +18,11 @@ class Origin:
     template_name: str | None
     def __init__(self, name: str, template_name: str | None) -> None: ...
 
+class Template:
+    template: Incomplete
+    backend: Jinja2
+    origin: Origin
+    def __init__(self, template: Incomplete, backend: Jinja2) -> None: ...
+    def render(self, context: Incomplete | None = ..., request: Incomplete | None = ...) -> Incomplete: ...
```
Comment (intgr) [inline django-stubs/template/backends/jinja2.pyi:26]: Is there a reason why these Jinja changes are in this PR? Seems like a separate topic

**Review by intgr — COMMENTED** [review]
Both changes look good to me, thanks, but I'd prefer if they were separate PRs

File: django-stubs/template/backends/jinja2.pyi (hunk 2 above)
Comment (sobolevn) [reply django-stubs/template/backends/jinja2.pyi:26]: Yes, new version of `pytest-mypy-plugins` requires `jinja2`. Now the environment has changed, new errors are detected by `stubtest`. See https://github.com/typeddjango/django-stubs/issues/1602

Comment (intgr) [reply django-stubs/template/backends/jinja2.pyi:26]: Oh I see. But we could merge splitted Jinja2 changes first and then this one separately.

Comment (intgr) [reply django-stubs/template/backends/jinja2.pyi:26]: Or if you don't think it's worth the effort, at least mention both changes in the PR title and commit message lead line

**Pull request merged**